src/esportsbot/cogs/MusicCog.py

```diff
@@ -0,0 +1,1117 @@
+        """
+
+        if not self._currently_active.get(guild_id) or len(self._currently_active.get(guild_id).get('queue')) == 0:
+            # If the queue is empty, provide the empty queue embed or the bot isn't active
+            updated_preview_message = self._no_current_song_message
+        else:
+            current_song = self._currently_active.get(guild_id).get('current_song')
+            updated_preview_message = Embed(title="Currently Playing: " + current_song.get('title'),
+                                            colour=Colour(0xd462fd), url=current_song.get('link'),
+                                            video=current_song.get('link'))
+            updated_preview_message.set_image(url=current_song.get('thumbnail'))
+
+        self.__add_time_remaining_field(guild_id, updated_preview_message)
+
+        return updated_preview_message
+
+    def __generate_link_data_from_queue(self, guild_id):
+        """
+        Get the stream data, thumbnails and other data regarding the top song in the queue.
+        :param guild_id: The guild id to get the queue from.
+        :return: A dict of information required to play a song.
+        """
+        current_song = self._currently_active.get(guild_id).get('queue')[0]
+        return self.__format_download_data(self.__download_video_info(current_song.get('link')))
+
+    def __download_link_list_info(self, links):
+        """
+        Downloads the basic information regarding a list of links.
+        :param links: The list of video links.
+        :return: A dict of each link with their video titles.
+        """
+        threads = min(20, len(links))
+        with concurrent.futures.ThreadPoolExecutor(max_workers=threads) as executor:
+            generator = executor.map(self.__get_basic_song_information, links)
+            return generator
+
+    async def __get_basic_song_information(self, url):
+        """
+        Download the title for a given url.
+        :param url: The url to find the title of.
+        :return: A dict with link and title keys.
+        """
+
+        # Using a random user agent means there is less chance of being blocked by google
+        user_agents = ['Mozilla/5.0 CK={} (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko',
+                       'Mozilla/5.0 (Windows NT 5.1; rv:7.0.1) Gecko/20100101 Firefox/7.0.1',
+                       'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) '
+                       + 'Chrome/74.0.3729.169 Safari/537.36',
+                       'Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.1; SV1; .NET CLR 1.1.4322)',
+                       'Mozilla/5.0 (Windows NT 6.1; WOW64; rv:40.0) Gecko/20100101 Firefox/40.1',
+                       'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) '
+                       + 'Chrome/74.0.3729.157 Safari/537.36']
+        shuffle(user_agents)
```

Comment:
Instead of:
```
        shuffle(user_agents)
        ...
        headers = {'User-Agent': user_agents[0], 'Referer': 'https://www.google.co.uk/'}
```

I would do:
```
        headers = {'User-Agent': random.choice(user_agents), 'Referer': 'https://www.google.co.uk/'}
```